src/main/java/org/opensearch/neuralsearch/query/HybridQueryScorer.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.query;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.Getter;
+
+import org.apache.lucene.search.DisiPriorityQueue;
+import org.apache.lucene.search.DisiWrapper;
+import org.apache.lucene.search.DisjunctionDISIApproximation;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.Scorer;
+import org.apache.lucene.search.Weight;
+
+/**
+ * Class abstracts functionality of Scorer for hybrid query. When iterating over documents in increasing
+ * order of doc id, this class fills up array of scores per sub-query for each doc id. Order in array of scores
+ * corresponds to order of sub-queries in an input Hybrid query.
+ */
+public final class HybridQueryScorer extends Scorer {
+
+    // score for each of sub-query in this hybrid query
+    @Getter
+    private final Scorer[] subScorers;
```

Comment:
Dose this have to be native array? Why not `ArrayList<Scorere>`?

Reply:
No particular reason, we do know number of elements (it's number of sub-queries) and arrays consume less resources. Any reason why do you think ArrayList will be better in this case?

Reply:
ArrayList provides better support for java stream operation.

Reply:
ok, let me try to refactor it to ArrayList